File: round_7/round_7.py
#N = casas e o numero de rounds
#X = A quantidade de casas puladas

#Entrada: Um número de casas N e uma lista com N inteiros representando o tamanho do pulo

#Você sempre começa na primeira casa

#Caso você ande para uma casa menor que 1 é o mesmo que ficar parado na casa 1

N = int(input())
lista = [int(input()) for _ in range(N)]
casa=1
chegou = False
for i in lista:
    casa +=i
    if casa < 1:
        casa = 1
    if casa >= N:
        chegou = True

if chegou:
    print('true')
else:
    print('false')

File: round_7/round_7.cpp
#include <iostream>
#include <list>
using namespace std;

int main(){
    int numero_rodadas, casa_atual, j;
    bool chegou;
    list<int> lista;

    casa_atual = 1;
    chegou = false;

    cin >> numero_rodadas;
    for(int i=0;numero_rodadas>i;i++){
        cin >> j;
        casa_atual += j;
        if (casa_atual < 1){
            casa_atual = 1;
        }
        if(casa_atual >= numero_rodadas){
            chegou = true;
        }
    }
    if (chegou==true){
        cout << "true";
    }
    else{
        cout << "false";
    }
}